### PR TITLE
Add custom formatting for TimeSpan that supports > 24Hr times

### PR DESCRIPTION
**Pull request by DaveyJonesBitPail**
## Description

TimeSpan formatting within the `AggregateStatisticsView` would show hours instead of total hours. This means that if we have more than 24 hours then we get Hours % 24 (example 30 hours timespan will only appear as if it was 6 hours).

This PR adds the following:
 - A convenience deconstructor extension to TimeSpan
 - A method within `AggregateStatisticsViewModel` to format a TimeSpan to a nice "intuitive" string like `31h 21m 34s`.
   - This intuitive string is composed based on TimeSpan length so for the following situations we get:
     - TimeSpan(days: 0, hours: 0, minutes:00, seconds: 30) : "30s"  
     - TimeSpan(days: 0, hours: 0, minutes:40, seconds: 30) : "40m 30s"
     - TimeSpan(days: 0, hours: 3, minutes:03, seconds: 05) : "3h 3m 5s"
     - TimeSpan(days: 1, hours: 1, minutes:00, seconds: 02) : "25h 0m 2s"

 - Changed `Duration` property within `AggregateStatisticsViewModel` to be formatted string rather than TimeSpan.
   - Perhaps controversial because now there is a slight "desync" between `AggregateStatistics` and `AggregateStatisticsViewModel`.
   - This is in lieu of an AvaloniaUI custom value converter.
 - Unit test to cover expected "nice formatting" within the view model
 - Unit test to cover deconstructor
 - Amend unit tests that relied on comparing `AggregateStatistics` to `AggregateStatisticsViewModel` to exclude `Duration` and have a separate UT to test formatting transform: `AggregateStatistics.Duration : TimeLapse` -> `AggregateStatisticsViewModel.Duration : string`

Fixes internally reported issue

### Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Refer to the added unit tests to see expected results  

### Reproduction steps

Process a file which takes more than 24 hours to compress, either via a very high compression level or a very large file.

## Checklist:

- [X] I followed the Contributing Guidelines.
- [X] I did a self-review.
- [X] I commented my code, particularly in hard-to-understand areas.
- [ ] I updated the documentation.
- [x] I updated the version of czicompress following [the README (Versioning)](../README.md#versioning) depending on the [type of change](#type-of-change)
  - Bug fix -> PATCH
  - New feature -> MINOR
  - Breaking change -> MAJOR
- [X] I have added tests that prove my fix is effective or that my feature works.
- [X] New and existing unit tests pass locally with my changes.

### CODE REVIEW

**Comment by DaveyJonesBitPail** [comment]
> Did you check that this works for an arbitrarily high number of hours? We can also move the string conversion to the view model (let's not introduce more converters).

Ah good point!

There is a limit to what we can store within timespan: TimeSpan.MaxValue which would look like `256204779:48:05:0000477` with the opinionated formatting. This is equivalent to what we expect "slightly more than 10,675,199 days" (source: https://learn.microsoft.com/en-us/dotnet/api/system.timespan.maxvalue?view=net-8.0&redirectedfrom=MSDN#remarks).

I'll add a test that is roughly equivalent to this.

**Comment by DaveyJonesBitPail** [comment]
> My guess would be that the original report was about
> 
> https://github.com/ZEISS/czicompress/blob/c986ba79eaab7c7295e5813bf935364e6590b71e/czishrink/netczicompress/Views/AggregateStatisticsView.axaml#L117
> 
> ```
> Text="{Binding Duration, Mode=OneWay, StringFormat='{}{0:hh\\:mm\\.ss}'}" />
> ```
> 
> Did you check that this works for an arbitrarily high number of hours? We can also move the string conversion to the view model (let's not introduce more converters).

Ah, I only looked at it from the logging perspective. I'll amend PR to also cover viewmodel.

**Comment by DaveyJonesBitPail** [comment]
> We can also move the string conversion to the view model (let's not introduce more converters).

We can move the string conversion here, but it's relevant to anything that uses CompressorMessage.FileFinished which also includes the CsvLogFileWriter. We either have this conversion there or use a timespan extension.

We can introduce days by default by making the format string in the viewmodel `{}{0:d\\:hh\\:mm\\.ss}` (I would just also update the Duration label to make this especially clear like: `Duration [d:hh:mm.ss]` or perhaps add a Tooltip). 

If we want to display days conditionally then we need a new Converter, no large issue with this either but in order to use it we link ViewModel logic within Model which I would want to avoid.

**Comment by DaveyJonesBitPail** [comment]
> I think I‘d prefer ‚30h 59m 59s‘ to ‚1:24:59:59‘ in the aggregate statistics (Btw, we might also have to fix the badge). For FileFinished, wouldn’t we need separate formatting anyways because we want higher precision there, at least when the timespan is short?

You're right, I've gone ahead and made something like this. The formatting function conditionally composes the "nicely readable" time depending on how long the elapsed time was. Here is an example with TimeStamp.Max:

mspaint_cnH6ZStZbu

and another with a significantly shorter time:
mspaint_XVtxw35C4e

`TimeSpan.ToString('c')` should remain sufficient for culture-invariant strings from timespan that **do** deal with situations > 24h (I was mistaken previously and had to read the doc a bit more closely). Therefore this change only affects the View + ViewModel

File: czishrink/netczicompress/Models/TimeSpanExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    /// <param name="value">TimeSpan to extend.</param>
+    /// <returns>Formatted date with TotalHours instead of Hours.</returns>
+    /// <remarks>
+    /// This is to solve the issue with <see cref="TimeSpan"/> that are longer than 24H. See https://stackoverflow.com/questions/39821991/how-to-display-hours-and-minutes-from-timespan-with-more-than-24-hours.
+    /// </remarks>
+    public static string ToDateTimeString(this TimeSpan value)
+        {
+        return string.Format(
+            "{0:00}:{1:00}:{2:00}:{3:0000000}",
+            value.TotalHours,
```
[the remark this answers is not in the document]
Comment (DaveyJonesBitPail) [reply czishrink/netczicompress/Models/TimeSpanExtensions.cs]: Ah you're right. Good way to truncate, thanks!

File: czishrink/netczicompress/AppComposer.cs [hunk 2]
```diff
@@ -72,7 +73,8 @@ public static MainViewModel ComposeMainViewModel(IProgramNameAndVersion info)
         var aggregateStatisticsViewModel = new AggregateStatisticsViewModel(
             aggregateIndicationViewModel: aggregateIndicationViewModel,
             ClipboardHelper.Instance,
```
Comment (m-aXimilian) [inline czishrink/netczicompress/AppComposer.cs:75]: Not related to this PR actually, but wouldn't it be more consistent to use the named argument style also for the ClipboardHelper?

Comment (DaveyJonesBitPail) [reply czishrink/netczicompress/AppComposer.cs:75]: Yes it would be. Sounds like a good excuse to create another PR 😄

**Pull request merged**